lib/LaTeXML/Package/listings.sty.ltxml

```diff
@@ -123,11 +123,11 @@ DefMacro('\lstinputlisting OptionalKeyVals:LST Semiverbatim', sub {
     my $text = listingsReadRawFile($gullet, $file);
     $STATE->getStomach->bgroup;
     lstActivate($keyvals);
-    my $name;
-    if (($name = lstGetTokens('name')) && !IsEmpty($name)) {
-      $file = $name; }
-    AssignValue('LST@toctitle', $file);    # so it shows up in list of..
-    return lstProcessDisplay($file, $text); });
+    my $name = lstGetTokens('name');
+    if (IsEmpty($name)) {
+      $name = $file; }
+    AssignValue('LST@toctitle', $name);    # so it shows up in list of..
```

Comment:
I'm thinking `$name=Invocation(T_CS('\texttt'),$name)` is necessary to get the verbatim tokens to digest right.  Then you don't need the other changes to `toctitle` or `\@@listings@block`

Reply:
Ah, I see.

Ok, refactored - works as expected. I still think we can remove the passthrough macro.

Reply:
Ah, but the `texttt` switches to typewriter font in a test case or two??

Reply:
Thanks, tests have been updated.

Reply:
Why is `texttt` being used here? After reading listings.sty again, the package keeps two copies of the name: the "internal" one is the list of tokens as read, the other is the same but with dollars, underscores, and hyphens replaced by corresponding text macros – not characters, to avoid font encoding problems. The latter form is what is used in captions and tocs (and for some odd bug, listings.sty does the replacement again when it loses track of the modified name). listings.sty does not switch font, at least not by default.

Reply:
Indeed, it would be more correct to introduce a new parameter type, similarly to what we've done with `HyperVerbatim` in the past, which is dedicated to this kind of listings treatment. `LstVerbatim` maybe? Then we won't need the font change.

Reply:
@xworld21 as you can see we're trying to juggle "brief", "expedient" and "correct" in finding a workable fix here. Not exactly the scientific method.

Reply:
If I understand the perl code correctly, one could also modify `lstActivate`, but that's less elegant, for sure.